src/transaction.ts

```diff
@@ -73,25 +74,51 @@ export interface KeyRange {
 }
 
 export interface ReadRequest extends RequestOptions {
-  keySet?: s.KeySet;
+  session?: string | null;
+  transaction?: TransactionSelector | null;
+  table?: string | null;
+  index?: string | null;
+  columns?: string[] | null;
   keys?: string[];
-  ranges?: KeyRange[];
-  index?: string;
-  columns?: string[];
-  limit?: number;
-  resumeToken?: ResumeToken;
-  partitionToken?: Uint8Array | string;
+  ranges?: spanner_client.spanner.v1.IKeyRange[];
+  keySet?: spanner_client.spanner.v1.IKeySet | null;
+  limit?: number | Long | null;
+  resumeToken?: Uint8Array | null;
+  partitionToken?: Uint8Array | null;
+}
+
+export interface TransactionSelector {
+  singleUse?: spanner_client.spanner.v1.ITransactionOptions | null;
+  id?: Uint8Array | string | null;
+  begin?: spanner_client.spanner.v1.ITransactionOptions | null;
 }
 
 export interface BatchUpdateError extends ServiceError {
   rowCounts: number[];
 }
 
+export interface CommitRequest {
+  session?: string | null;
+  transactionId?: Uint8Array | string | null;
+  singleUseTransaction?: spanner_client.spanner.v1.ITransactionOptions | null;
+  mutations?: spanner_client.spanner.v1.IMutation[] | null;
+}
+
 export type BatchUpdatePromise = Promise<[number[], s.ExecuteBatchDmlResponse]>;
-export type BeginPromise = Promise<[s.Transaction]>;
-export type CommitPromise = Promise<[s.CommitResponse]>;
+export type BeginPromise = Promise<[spanner_client.spanner.v1.ITransaction]>;
+export interface BeginTransactionCallback {
+  (
+    err: Error | null,
+    transaction:
+      | spanner_client.spanner.v1.ITransaction
+      | spanner_client.spanner.v1.ISession
```

Comment:
is `ISession` returned anywhere?

Reply:
don't required `spanner_client.spanner.v1.ISession` Removed!!